### ISSUE TITLE
New tag controls store data in umbracoPropertyData.varcharValue not textValue

### ISSUE THREAD

**Issue by agrath** [umbraco/Umbraco-CMS#8468]
The default tag control set to JSON mode serializes to data that looks similar to the following

```
["Doing business in new zealand","Overseas Investment Office","OIO","regulatory consents","mergers and acquisitions","M&A","joint ventures","JV","establishing companies and subsidiaries","cross-border","internal tax implications","foreign organisations","real estate","commercial property","Māori law","rules and regulations","international organisation regulations","Maori business","Chinese businesses","resource management","RMA","Treaty of Waitangi","capital markets","OIO applications"]
```

This data gets saved to umbracoPropertyData.varcharValue for *newly created tag editors* which means that there is an implicit limit on the total length of tag data of 512 characters (including JSON serialization overhead)

## Umbraco version

This issue was reported to me on Umbraco version: 8.4.0
I have also reproduced it on 8.7.0

Reproduction
------------
Create a new tag editor property (e.g. specifying the group) then fill out a bunch of tag content, broken out into multiple tags. 

At a point (depending on how many tags and how long they are) the page will start throwing string or binary data would be truncated when doing a save and publish.

This is similar to my previous issue https://github.com/umbraco/Umbraco-CMS/issues/8412 however in this case, the length is exceeded across multiple tags, not just one.

### Bug summary

A newly created tag datatype is configured to store data in the in the wrong umbracoPropertyData field/column, limiting the length/number of tags allowed for that property.

Here is a video showing a new property being added with the resulting dataType getting the wrong dbType value
[ice_video_20200724-142535.zip](https://github.com/umbraco/Umbraco-CMS/files/4969937/ice_video_20200724-142535.zip)

(see sql view)

### Specifics

On a page containing the tag control, add a bunch of tags, e.g. as per this screenshot

![image](https://user-images.githubusercontent.com/78284/88356051-ab505080-cdba-11ea-9119-67f41bb70529.png)

When you add another tag ("fourty-five") and exceed 512 characters for *serialized* data then you won't be able to publish the page any more.
This was reported to me in an 8.4.0 site and I have also tested it on 8.7.0 with the same result

The tag tests in ContentServiceTagsTests appears to test using ValueStorageType.Ntext rather than ValueStorageType.Varchar  and I would expect the data to be stored in Ntext

In my specific case, the umbracoDataType for the default tag editor (id 1041) has dbType set to Ntext but a subsequent tag editor that has been created (1080) appears to have the dbType set to nvarchar causing the issue.

### Expected result

Able to save a node with long tags

### Actual result

Get a string or binary data truncated error

![image](https://user-images.githubusercontent.com/78284/88356024-97a4ea00-cdba-11ea-8155-7ac5c142ba1d.png)

### Workaround

The following SQL will migrate the broken data, do a deep publish after

```
update umbracodatatype set dbtype = 'Ntext' where propertyEditorAlias = 'Umbraco.Tags'
update umbracoPropertyData set textvalue = varcharValue where propertyTypeId in (select id from cmsPropertyType where dataTypeId in (select nodeid from umbracoDataType where propertyEditorAlias = 'Umbraco.Tags')) and textvalue is null and varcharvalue is not null
update umbracopropertydata set varcharvalue = null where propertyTypeId in (select id from cmsPropertyType where dataTypeId in (select nodeid from umbracoDataType where propertyEditorAlias = 'Umbraco.Tags')) and textvalue is not null and varcharvalue is not null
```

To summarize, the issue appears to be when the new datatype (for a property) is created the resulting umbracodatatype database record gets the wrong dbType

**Comment by rbottema** [comment]
See also #6961.

**Comment by agrath** [comment]
Many thanks @rbottema 
I didn't see that issue when I searched, so that issue should definitely be linked across.

It looks like we both arrived at the same fix and I have also noted that this may be a breaking change in terms of data being in the wrong column after the change.

I have tested my migration script above to copy the data on a site we're building and confirm the tag control continued to work with both this patch and the data copied (e.g. the tags moved and continued to work), perhaps this PR should be adapted to include a migration to migrate both the storage type and copy the data as per my script?

I don't know how to add this migration though, I've not done that before - how do we do it?
